dbms/src/Storages/DeltaMerge/StoragePool.h

```diff
@@ -173,7 +174,16 @@ class StoragePool : private boost::noncopyable
 
     PageIdU64 newDataPageIdForDTFile(StableDiskDelegator & delegator, const char * who);
     PageIdU64 newLogPageId() { return ++max_log_page_id; }
-    PageIdU64 newMetaPageId() { return ++max_meta_page_id; }
+    PageIdU64 newMetaPageId()
+    {
+        // Reserve the last page id for encryption key
+        RUNTIME_CHECK_MSG( // NOLINT
+            ns_id < std::numeric_limits<UInt64>::max() || max_meta_page_id < std::numeric_limits<UInt64>::max(),
+            "ns_id {}, page_id {} are reserved for keyspace encryption key",
+            ns_id,
+            max_meta_page_id);
+        return ++max_meta_page_id;
+    }
```

Comment:
Does the encryption key generated and stored for each `physical_table_id` instead of `keyspace_id`?

Reply:
No, each `keyspace_id` has only one encryption key.

The format of page id is `[optional KeyspaceID] + "tm" + NamespaceID + PageIDU64`, just keep it simple.